File: lotteryticket/src/main/java/com/wuhai/lotteryticket/contract/base/INormalRxLifecycleViewInterFace.java
package com.wuhai.lotteryticket.contract.base;

import com.trello.rxlifecycle.LifecycleTransformer;

/**
 * view 基类
 */
public interface INormalRxLifecycleViewInterFace {

    /**
     * 加载loading
     */
    void showLoading();

    /**
     * 隐藏loading
     */
    void dimssLoading();

    /**
     * ac 销毁时候，取消订阅
     * @return
     */
    LifecycleTransformer getLifecycleTransformer();
}

File: lotteryticketkotlin/src/main/java/com/wuhai/demo/lotteryticketkotlin/contract/base/INormalRxLifecycleViewInterFace.kt
package com.wuhai.demo.lotteryticketkotlin.contract.base

import com.trello.rxlifecycle.LifecycleTransformer

/**
 * view 基类
 */
interface INormalRxLifecycleViewInterFace {
    /**
     * 加载loading
     */
    fun showLoading()

    /**
     * 隐藏loading
     */
    fun dimssLoading()

    /**
     * ac 销毁时候，取消订阅
     * @return
     */
    val lifecycleTransformer: LifecycleTransformer<*>?
}